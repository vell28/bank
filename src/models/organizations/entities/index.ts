export * from './account';
export * from './balance';
export * from './client';
export * from './organization';
