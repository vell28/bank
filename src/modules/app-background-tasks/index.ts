import { idleLogout } from './idleLogout';
import { backgroundUpdateAccountsBalance, backgroundUpdateAccountTask } from './accounts';

// TODO: add here loading profile, loading history, etc..
export default {
  accounts: backgroundUpdateAccountTask(),
  accountsBalance: backgroundUpdateAccountsBalance(),
  idle: idleLogout(),
};
