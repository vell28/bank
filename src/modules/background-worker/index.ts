import { forEachObjIndexed } from 'ramda';
import {
  ITask, BackgroundTask, ITaskParams, ITaskResponse, taskState
} from './task';

interface ITaskMap {
  [key: number]: ITask;
}

interface IBackgroundWorker {
  createTask: (params: ITaskParams) => ITaskResponse;
  removeTask: (id: number) => ITaskResponse;
  forceTask: (id: number) => ITaskResponse;
  removeAllTask: () => void;
}

class BackgroundWorker implements IBackgroundWorker {
  private readonly tasks: ITaskMap = {};

  public createTask(params: ITaskParams) {
    const task = new BackgroundTask(params);
    const resp = task.start();
    this.tasks[resp.id] = task;
    return resp;
  }

  public removeTask(id: number) {
    const task = this.tasks[id];
    if (!task) {
      return this.createNotFoundError(id);
    }
    delete this.tasks[id];

    return task.remove();
  }

  public forceTask(id: number) {
    const task = this.tasks[id];
    if (!task) {
      return this.createNotFoundError(id);
    }
    return task.force();
  }

  public removeAllTask() {
    forEachObjIndexed((task: ITask) => {
      task.remove();
    }, this.tasks);
  }

  // eslint-disable-next-line class-methods-use-this
  private createNotFoundError(id: number): ITaskResponse {
    return {
      state: taskState.TERMINATED,
      id,
      executeCount: -1,
    };
  }
}

export default new BackgroundWorker();
