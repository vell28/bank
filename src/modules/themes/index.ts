import { IThemeColors, colorPalette } from './colorPalettes';
import {
  SMALL_VIEWPORT, MEDIUM_VIEWPORT, LARGE_VIEWPORT, MOBILE_VIWEPORT, fonts
} from './config';

export interface ITheme {
  name: string;
  colors: IThemeColors;
  dimension: object;
  size: string;
  breakpoints: string[];
  fonts: {
    faFontsPath: string;
    fontFamily: string;
  };
  typography: {
    bodyBackground: string;
    bodyFontColor: string;
    bodyFontFamily: string;

    globalFontSize: string;
    globalWeightNormal: number;
    globalLineHeight: number;

    headersFontColor: string;
    headersFontFamily: string;
    headersFontWeight: number;
  };
}

export const mainDarkTheme: ITheme = {
  name: 'main-dark-theme',
  colors: colorPalette,
  dimension: {},
  size: 'default',
  breakpoints: [`${MOBILE_VIWEPORT}px`, `${SMALL_VIEWPORT}px`, `${MEDIUM_VIEWPORT}px`, `${LARGE_VIEWPORT}px`],
  fonts,
  typography: {
    bodyBackground: colorPalette.darkBlue,
    bodyFontColor: colorPalette.white,
    bodyFontFamily: fonts.fontFamily,

    globalFontSize: '1.6rem',
    globalWeightNormal: 400,
    globalLineHeight: 1.5,

    headersFontColor: colorPalette.white,
    headersFontFamily: fonts.fontFamily,
    headersFontWeight: 700,
  },
};
