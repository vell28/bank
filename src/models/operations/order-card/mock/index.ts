import { IServerAddress } from 'models/organizations/redux/actions/entities';
import { IClient } from 'models/organizations/entities';
import { IAddress, ICardOrderOwner } from '../entities';

export const exampleAddress: IAddress = {
  countryCode: 'DE',
  city: 'Cologne',
  postalCode: '50667',
  street: 'Domkloster',
  streetNumber: '4',
  apartmentNumber: 0,
};

export const exampleOrganizationAddress: IAddress = {
  countryCode: 'DE',
  city: 'Cologne',
  postalCode: '50667',
  street: 'Domkloster',
  streetNumber: '4',
  apartmentNumber: 0,
};

export const exampleClientInfo: ICardOrderOwner = {
  embossedName: 'Frank-Walter Steinmeier',
  phoneNumber: '9111111',
};

export const exampleServerAddress: IServerAddress = {
  country: 'Germany',
  province: 'North Rhine-Westphalia',
  city: 'Düsseldorf',
  postalCode: '40479',
  street: 'Gartenstraße',
  streetNumber: '53',
  apartmentNumber: '42',
};

export const exampleClient: IClient = {
  address: exampleOrganizationAddress,
  name: 'Frank-Walter Steinmeier',
  phoneNumber: '9111111',
};
