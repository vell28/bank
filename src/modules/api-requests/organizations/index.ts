import { IUpdateBalanceResponse } from 'models/organizations/entities';
import { IServerOrganization } from 'models/organizations/redux/actions/entities';
import { makeGetRequest } from '../../api-client/request';

import { ResponseAPI } from '../../api-client/entities';

import { host, apiVersion } from '../const';

export const fetchAccountBalanceRequest = (accountId: string): Promise<ResponseAPI<IUpdateBalanceResponse>> =>
  makeGetRequest<IUpdateBalanceResponse>(`${host}/api/${apiVersion}/accounts/${accountId}/balances`, {});

export const fetchOrganizationRequest = (): Promise<ResponseAPI<IServerOrganization[]>> =>
  makeGetRequest<IServerOrganization[]>(`${host}/api/v2/organizations`, {});
