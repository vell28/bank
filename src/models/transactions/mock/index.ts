import { CurrencyCodes } from 'modules/currencies';
import { ITransaction, ITransactionResponse } from '../entities';

export const transactionData: ITransaction = {
  id: 'PPY184598',
  amount: -10.0,
  fee: 0,
  status: 'COMPLETED',
  paymentType: 'TRANSFER',
  fromNumber: 'MT98PAPY36836000002676370004519',
  fromName: 'Ralf Williams',
  fromBank: 'Papaya',
  toNumber: 'MT97PAPY36836000002676370003470',
  toName: 'Dmitrii Popov',
  toBank: 'Papaya',
  currency: {
    code: CurrencyCodes.EUR,
  },
  clientId: '6449',
  accountId: 'PPY5107',
  description: 'To: Dmitrii Popov Vvg',
  executedAt: '2019-10-11T10:12:00',
  referenceNumber: 'PPY1e1d31e3-6acb-4d60-9f70-b9151221d11',
  paymentToRepeat: {
    type: 'PAYMENT_PHONE',
    fromPhone: '79111111111',
    account: 'PPY5107',
    fromCardPan: null,
    fromCardId: null,
    beneficiaryName: null,
    beneficiaryAccount: null,
    country: null,
    city: null,
    address: null,
    swiftCode: null,
    beneficiaryBank: null,
    intermediaryBank: null,
    intermediarySwift: null,
    intermediaryAccount: null,
    commissionType: null,
    urgency: null,
    iban: null,
    phoneNumber: '79085491121',
    cardNumber: null,
    currencyCode: CurrencyCodes.EUR,
    amount: 10,
    purpose: 'other Vvgh',
    transferDetails: null,
  },
  openapiData: {
    type: 'TO_CONTACT',
    toCard: null,
    toPhone: '79085491121',
    subType: null,
    externalOwnerID: null,
    fromCardPan: null,
  },
  operationType: null,
  fromPanDisplay: null,
};

const second: ITransaction = {
  id: 'PPY184597',
  amount: -1.0,
  fee: 0,
  status: 'COMPLETED',
  paymentType: 'TRANSFER',
  fromNumber: 'MT98PAPY36836000002676370004519',
  fromName: 'Ralf Williams',
  fromBank: 'Papaya',
  toNumber: 'MT52PAPY36836000002676370000303',
  toName: 'Elfur Long',
  toBank: 'Papaya',
  currency: {
    code: CurrencyCodes.EUR,
  },
  clientId: '6449',
  accountId: 'PPY5107',
  description: 'To: Elfur Long 12',
  executedAt: '2019-10-11T10:08:47',
  referenceNumber: 'PPY84f7087c-c511-4928-b299-13ee47c1bd9',
  paymentToRepeat: {
    type: 'PAYMENT_PHONE',
    fromPhone: '79111111111',
    account: 'PPY5107',
    fromCardPan: null,
    fromCardId: null,
    beneficiaryName: null,
    beneficiaryAccount: null,
    country: null,
    city: null,
    address: null,
    swiftCode: null,
    beneficiaryBank: null,
    intermediaryBank: null,
    intermediarySwift: null,
    intermediaryAccount: null,
    commissionType: null,
    urgency: null,
    iban: null,
    phoneNumber: '79032950311',
    cardNumber: null,
    currencyCode: CurrencyCodes.EUR,
    amount: 1,
    purpose: 'other 123',
    transferDetails: null,
  },
  openapiData: {
    type: 'TO_CONTACT',
    toCard: null,
    toPhone: '79032950311',
    subType: null,
    externalOwnerID: null,
    fromCardPan: null,
  },
  operationType: null,
  fromPanDisplay: null,
};

export const transactionList = [transactionData, second];

export const transactionResponse: ITransactionResponse = {
  content: transactionList,
  maxPageNumber: 10,
  pageSize: 20,
  pageNumber: 1,
  length: 0,
};
