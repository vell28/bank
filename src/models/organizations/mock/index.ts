import { CurrencyCodes } from 'modules/currencies';
import {
  IAccount, AccountStatus, AccountTypes, AccountColors
} from '../entities';
import { ICard, CardProduct, Limit } from '../../operations/card-settings/entities';

import { IServerClient } from '../redux/actions/entities';
import { exampleServerAddress } from '../../operations/order-card/mock';

export const cardData: ICard = {
  accountId: 'PPY5107',
  availableBalance: 1897.61,
  balance: 0,
  balanceHold: 0,
  balances: [
    {
      currency: CurrencyCodes.EUR,
      balance: 1030.05,
      balanceHold: 0,
      availableBalance: 1897.61,
    },
  ],
  clientId: '6449',
  currency: { code: CurrencyCodes.EUR },
  expireAt: '2022-05-01T00:00:00',
  id: 'PPY13643',
  limits: [
    {
      type: Limit.DAY,
      title: null,
      maxValue: 500,
      usedValue: 0,
    },
  ],
  number: '5277 60_3 984',
  options: {
    limits: { disable: false },
  },
  owner: { embossedName: 'RALF WILLIAMS' },
  paySystem: 'MASTERCARD',
  plasticStatus: 'PENDING_SHIPMENT',
  productType: CardProduct.MAIN,
  status: AccountStatus.ACTIVE,
};

export const accountsData: IAccount[] = [
  {
    accountName: 'MT98PAPY36836000002676370004519',
    accountType: AccountTypes.PHYSICAL,
    accruedInterest: null,
    availableBalance: 1897.61,
    availableOperations: [],
    balance: 0,
    balanceHold: 0,
    color: AccountColors.DEFAULT,
    balances: [
      {
        currency: CurrencyCodes.EUR,
        balance: 1030.05,
        balanceHold: 0,
        availableBalance: 1897.61,
      },
    ],
    bonusProgramActivate: false,
    clientId: '6449',
    createdAt: '2019-11-24T14:59:55',
    currency: { code: CurrencyCodes.EUR },
    id: 'PPY5107',
    number: 'MT98PAPY36836000002676370004519',
    status: AccountStatus.ACTIVE,
  },
  {
    accountName: 'MT98PAPY36836000002676370004519',
    accountType: AccountTypes.JURIDICAL,
    accruedInterest: null,
    availableBalance: 1897.61,
    availableOperations: [],
    balance: 0,
    balanceHold: 0,
    color: AccountColors.DEFAULT,
    balances: [
      {
        currency: CurrencyCodes.EUR,
        balance: 1030.05,
        balanceHold: 0,
        availableBalance: 1897.61,
      },
    ],
    bonusProgramActivate: false,
    clientId: '6449',
    createdAt: '2019-11-24T14:59:55',
    currency: { code: CurrencyCodes.EUR },
    id: 'PPY6666',
    number: 'MT98PAPY36836000002676370004519',
    status: AccountStatus.ACTIVE,
  },
];

export const clientData: IServerClient = {
  id: '42',
  name: 'Test Client',
  citizenship: 'Germany',
  email: 'test.client@test.com',
  phoneNumber: '222333',
  address: exampleServerAddress,
};
