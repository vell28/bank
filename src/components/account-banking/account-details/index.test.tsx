import React from 'react';

import { ThemeProvider } from 'styled-components';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import { mainDarkTheme } from 'modules/themes';
import { accountsData } from 'models/organizations/mock';
import { AccountDetails } from '.';

describe('<AccountDetails /> tests', () => {
  test('should success render', () => {
    const component: any = render(
      <ThemeProvider theme={mainDarkTheme}>
        <AccountDetails account={accountsData[0]} />
      </ThemeProvider>,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
