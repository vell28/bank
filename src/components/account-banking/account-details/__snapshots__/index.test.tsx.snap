// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountDetails /> tests should success render 1`] = `
<div
  class="sc-AxirZ itToNK"
>
  <div
    class="sc-AxiKw jiRFHs"
  >
    IBAN: MT98PAPY36836000002676370004519
  </div>
  <div
    class="sc-AxhCb cARexh"
  >
    <div
      class="sc-AxhUy lbQxAs"
    >
      <div
        class="sc-AxgMl fHVZmz"
      >
        The beneficiarys bank
      </div>
      <div
        class="sc-AxgMl sc-AxheI gwEExQ"
      />
    </div>
    <div
      class="sc-AxhUy lbQxAs"
    >
      <div
        class="sc-AxgMl fHVZmz"
      >
        BIC
      </div>
      <div
        class="sc-AxgMl sc-AxheI gwEExQ"
      />
    </div>
  </div>
  <div
    class="sc-Axmtr fsTIDm"
  >
    <button
      class="sc-AxjAm sc-AxmLO lgoOxM"
    >
      copy
    </button>
    <button
      class="sc-AxjAm sc-AxmLO lgoOxM"
    >
      export
    </button>
  </div>
</div>
`;
